pandas_gbq/gbq.py

```diff
@@ -188,78 +172,103 @@ def get_application_default_credentials(self):
             from the environment. Or, the retrieved credentials do not
             have access to the project (self.project_id) on BigQuery.
         """
-        import httplib2
-        try:
-            from googleapiclient.discovery import build
-        except ImportError:
-            from apiclient.discovery import build
-        try:
-            from oauth2client.client import GoogleCredentials
-        except ImportError:
-            return None
+        credentials, _ = google.auth.default(scopes=[self.scope])
+        return self.try_credentials(credentials)
 
-        try:
-            credentials = GoogleCredentials.get_application_default()
-        except:
-            return None
+    def load_user_account_credentials(self):
+        """
+        Loads user account credentials from a local file.
 
-        http = httplib2.Http()
-        try:
-            http = credentials.authorize(http)
-            bigquery_service = build('bigquery', 'v2', http=http)
-            # Check if the application has rights to the BigQuery project
-            jobs = bigquery_service.jobs()
-            job_data = {'configuration': {'query': {'query': 'SELECT 1'}}}
-            jobs.insert(projectId=self.project_id, body=job_data).execute()
-            return credentials
-        except:
-            return None
+        Parameters
+        ----------
+        None
 
-    def get_user_account_credentials(self):
-        from oauth2client.client import OAuth2WebServerFlow
-        from oauth2client.file import Storage
-        from oauth2client.tools import run_flow, argparser
+        Returns
+        -------
+        - GoogleCredentials,
+            If the credentials can loaded. The retrieved credentials should
+            also have access to the project (self.project_id) on BigQuery.
+        - OR None,
+            If credentials can not be loaded from a file. Or, the retrieved
+            credentials do not have access to the project (self.project_id)
+            on BigQuery.
+        """
+        with open('bigquery_credentials.dat') as credentials_file:
```

Comment:
this will hard fail with an odd message if this file doesn't exist. is there a reason this is hardcoded? (not to mention pathed), better to have its location specified by argument.